flocker/control/httpapi.py

```diff
@@ -492,11 +498,15 @@ def create_container_configuration(self, host, name, image, ports=()):
                 external_port=port['external']
             ))
 
+        if environment:
```

Comment:
I agree that json schema validation should prevent anything other than a dictionary here, but I still think it's better to be explicit about what's expected.